src/writeFile.js

```diff
@@ -23,7 +26,34 @@ export default function writeFile(globalRef, pattern, file) {
         return fs.readFileAsync(file.absoluteFrom)
         .then((content) => {
             if (pattern.transform) {
-                content = pattern.transform(content, file.absoluteFrom);
+                const transform = (content, absoluteFrom) => {
+                    return pattern.transform(content, absoluteFrom);
+                };
+
+                if (pattern.cache) {
+                    const cacheKey = pattern.cacheKey
+                        ? pattern.cacheKey
+                        : JSON.stringify(serialize({
```

Comment:
Could we use/generate an `id` or `hash`  for patterns to avoid serialization ?

Reply:
@michael-ciniawsky we can generate md5 or sha algo for this

Reply:
@michael-ciniawsky after testing with difference arguments and platform, seems `serialize` is best solution, btw we can do this in `uglify` plugin